bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/KNXChannelType.java

```diff
@@ -173,11 +174,17 @@ protected final Set<String> asSet(String... values) {
                 if (expectedTypeClass != null) {
                     if (expectedTypeClass.isInstance(command)) {
                         logger.trace(
-                                "getCommandSpec key '{}' uses expectedTypeClass '{}' witch isInstance for command '{}' and dpt '{}'",
+                                "getCommandSpec key '{}' uses expectedTypeClass '{}' which isInstance for command '{}' and dpt '{}'",
                                 key, expectedTypeClass, command, dpt);
                         return new WriteSpecImpl(config, dpt, command);
                     }
                 }
+                if (command instanceof QuantityType) {
```

Comment:
Hm, is it really correct to do an independent "if" here and not check at all the dpt anymore?
I would expect the `typeHelper` in line 173 to return `DecimalType` as the `expectedTypeClass`.
If this is the case, we would probably only need to adapt line 175 to
```
                    if (expectedTypeClass.isAssignableFrom(command.getClass())) {
```
so that it also evaluates to true, if `command` is a QuantityType and not only if it is a DecimalType.

Wdyt? Am I missing something?

Reply:
@kaikreuzer I just did a quick check: ```expectedTypeClass``` returns ```DecimalType```, ```command.getClass()```is ```QuantityType``` and ```isAssignableFrom``` returns false.

I don't think that we nedd to check the dpt - the type conversion will anyway only work if the unit conversion is possible.

Reply:
You're right, I forgot that `QuantityType` does not extend `DecimalType`. 😞 
I would nonetheless prefer to do a proper check of the type in the if clause.
If `expectedTypeClass` is null, we should never proceed, so the code should imho move into the if statement above.
And then we could simply enhance line 175 to
```
                    if (expectedTypeClass.isInstance(command) || (expectedTypeClass==DecimalType.class && command instanceof QuantityType<?>)) {
```
(and slightly change the tracing message).

Reply:
This looks reasonable, updated code as requested.
Thanks Kai!